### DISCUSSION TITLE
Discussion - third party Windows apps need a recommended way to open files in WSL

### DISCUSSION THREAD

**Discussion by mikemaccana** [microsoft/WSL#6234] [General]
**Is your feature request related to a problem? Please describe.**
I use Windows git tools to manage my changes. I frequently have my project open in VScode using WSL. However when I open a file in the Window git tool, it doesn't use WSL, therefore opening a second window:

![image](https://user-images.githubusercontent.com/172594/99064978-9d5b5100-259e-11eb-9ba1-a2b4c064f0f0.png)

**Describe the solution you'd like**
I'd like it if somehow the third party app used the existing VScode window.

**Describe alternatives you've considered**

 - VScode would somehow work out these were the same files
 - The third party Windows app could follow some kind of guidelines from Microsoft to make an 'open with WSL' button

**Additional context**
The key thing here is I (as a user) am not sure what I should ask my software vendor for. Eg, should vscode recognise the change? Or should my tool (git fork) have an 'open in WSL' button? 

If there was some published official guidance on how third party apps should work with WSL it would be very useful.

**Comment by therealkenc** [comment]
Documentation on WSL interop is [here](https://docs.microsoft.com/en-us/windows/wsl/interop). There is also an api documented [here](https://docs.microsoft.com/en-us/windows/win32/api/_wsl/).

> VScode would somehow work out these were the same files

If they are the same files they are the same files, and VS Code (or anything else) will 'recognize' them as such. But heads up #4739 and sister issue #4581 (depending on the direction). WSL2 is presently does not support file watching.

**Comment by mikemaccana** [comment]
> If they are the same files they are the same files, and VS Code (or anything else) will 'recognize' them as such.

Can I confirm that when I open a file via a Windows path, if VScode already has the parent folder opened via WSL, that the file opened via a Windows path should appear in the existing window with the parent opened via WSL?

That's not happening in my test machine - I get the file opened via a Windows path in a new window with  a differently colored status bar.

If that's the intended behavior it might be better for me to convert this to a VScode issue.
